main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicSourceManager.java

```diff
@@ -167,13 +180,37 @@ private AudioItem getPlaylist(String userString, String id) throws IOException {
 		return new YandexMusicAudioPlaylist(playlistTitle, tracks, ExtendedAudioPlaylist.Type.PLAYLIST, json.get("result").get("url").text(), this.formatCoverUri(coverUri), author);
 	}
 
+	private List<JsonBrowser> getTracks(String trackIds) throws URISyntaxException, IOException {
+		return getJson(PUBLIC_API_BASE + "/tracks", new ArrayList<>() {{
+			add(new BasicNameValuePair("track-ids", trackIds));
+		}}).get("result").values();
+	}
```

Comment:
this is some cursed code
```suggestion
	private List<JsonBrowser> getTracks(String trackIds) throws IOException {
		return getJson(PUBLIC_API_BASE + "/tracks?track-ids=" + URLEncoder.encode(trackIds, StandardCharsets.UTF_8)).get("result").values();
	}
```

this gets rid off like 3 other fucntions